src/utils/openai.ts

```diff
@@ -123,54 +125,56 @@ const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 // };
 
 export const generateCommitMessage = async (
-	apiKey: string,
-	model: TiktokenModel,
-	locale: string,
-	diff: string,
-	completions: number,
-	maxLength: number,
-	type: CommitType,
-	timeout: number,
-	proxy?: string,
+  apiKey: string,
+  model: TiktokenModel,
+  locale: string,
+  diff: string,
+  completions: number,
+  maxLength: number,
+  type: CommitType,
+  timeout: number,
+  proxy?: string,
+  baseUrl?: string,
 ) => {
-	try {
-		const completion = await createChatCompletion(
-			apiKey,
-			{
-				model,
-				messages: [
-					{
-						role: 'system',
-						content: generatePrompt(locale, maxLength, type),
-					},
-					{
-						role: 'user',
-						content: diff,
-					},
-				],
-				temperature: 0.7,
-				top_p: 1,
-				frequency_penalty: 0,
-				presence_penalty: 0,
-				max_tokens: 200,
-				stream: false,
-				n: completions,
-			},
-			timeout,
-			proxy,
-		);
+  try {
+    const completion = await createChatCompletion(
+      apiKey,
+      {
+        model,
+        messages: [
+          {
+            role: 'system',
+            content: generatePrompt(locale, maxLength, type),
+          },
+          {
+            role: 'user',
+            content: diff,
+          },
+        ],
+        temperature: 0.7,
+        top_p: 1,
+        frequency_penalty: 0,
+        presence_penalty: 0,
+        max_tokens: 200,
+        stream: false,
+        n: completions,
+      },
+      timeout,
+      proxy,
+      baseUrl,
+    );
 
-		return deduplicateMessages(
-			completion.choices
-				.filter(choice => choice.message?.content)
-				.map(choice => sanitizeMessage(choice.message!.content)),
-		);
-	} catch (error) {
-		const errorAsAny = error as any;
-		if (errorAsAny.code === 'ENOTFOUND') {
-			throw new KnownError(`Error connecting to ${errorAsAny.hostname} (${errorAsAny.syscall}). Are you connected to the internet?`);
-		}
+    return deduplicateMessages(
+      completion.choices
+        .filter(choice => choice.message?.content)
+        .map(choice => sanitizeMessage(choice.message!.content)),
+    );
+  } catch (error) {
+    const errorAsAny = error as any;
```

Comment:
Is it possible to not cast `error` to `any`? What do you think about something like this?
```typescript
if ("code" in "error" && error.code === "ENOTFOUND") {
// ...
}
```